### PR TITLE
[flutter_tools] reland remove flutter view cache

### LINKED ISSUE #56194: Error connecting to the service protocol: No Flutter view is available on "device_name"
## Steps to Reproduce

1. Trying to run flutter app after recent update in master channel with `flutter run -v`
2. It installs the APK but gets disconnected with the following error message.
`Error connecting to the service protocol: No Flutter view is available on "device_name"`

`flutter doctor` output:
```
[√] Flutter (Channel master, 1.18.0-9.0.pre.93, on Microsoft Windows [Version 10.0.18362.720], locale en-IN)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Android Studio (version 3.5)
[√] VS Code (version 1.44.2)
[√] Connected device (3 available)

• No issues found!
```
## Logs

```
[ +190 ms] Successfully connected to service protocol: http://127.0.0.1:54365/RAoQ3a4zxzY=/
[   +7 ms] Error connecting to the service protocol: No Flutter view is available on ONEPLUS A6000.
[   +4 ms] "flutter run" took 57,058ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:570:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:796:18)
#3      _rootRunUnary (dart:async/zone.dart:1192:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:526:5)
#9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:303:13)
#11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#12     _rootRunUnary (dart:async/zone.dart:1192:38)
#13     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#14     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#16     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#17     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:303:13)
#20     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#21     _rootRunUnary (dart:async/zone.dart:1192:38)
#22     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#23     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#25     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#26     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#27     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:560:7)
#28     _rootRun (dart:async/zone.dart:1184:13)
#29     _CustomZone.run (dart:async/zone.dart:1077:19)
#30     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#32     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)

```

```
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes #56194

Remove caching of FlutterView and poll forever if the list of flutter views is empty.

Added test for missing release mode logic.

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -226,38 +226,13 @@ class FlutterDevice {
...
-      ];
-    }
-    return _views;
-  }
-
   Future<void> exitApps({
     @visibleForTesting Duration timeoutDelay = const Duration(seconds: 10),
   }) async {
-    if (!device.supportsFlutterExit) {
+    if (!device.supportsFlutterExit || vmService == null) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/resident_runner.dart:232]: This is the fix

File: packages/flutter_tools/test/general.shard/resident_runner_test.dart [hunk 2]
```diff
@@ -811,10 +818,20 @@ void main() {
     expect(fakeVmServiceHost.hasRemainingExpectations, false);
   }));
 
+  test('FlutterDevice can exit from a release mode isolate with no VmService', () => testbed.run(() async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/resident_runner_test.dart:821]: This is the test

**Comment by jonahwilliams** [comment]
NNBD when!

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**